src/cookieSyncManager.ts

```diff
@@ -0,0 +1,184 @@
+    this: ICookieSyncManager,
+    mpInstance: MParticleWebSDK
+) {
+    const self = this;
+
+    // Public
+    this.attemptCookieSync = (
+        previousMPID: MPID,
+        mpid: MPID,
+        mpidIsNotInCookies: boolean
+    ): void => {
+        if (!mpid || mpInstance._Store.webviewBridgeEnabled) {
+            return;
+        }
+
+        const { pixelConfigurations } = mpInstance._Store;
+        const persistence = mpInstance._Persistence.getPersistence();
+
+        pixelConfigurations.forEach(pixelSettings => {
+            // set requiresConsent to false to start each additional pixel configuration
+            // set to true only if filteringConsenRuleValues.values.length exists
+            let requiresConsent = false;
+
+            // Filtering rules as defined in UI
+            const { filteringConsentRuleValues } = pixelSettings;
+            const { values } = filteringConsentRuleValues || {};
+
+            if (!isEmpty(values)) {
+                requiresConsent = true;
+            }
+
+            // Kit Module ID
+            const moduleId = pixelSettings.moduleId.toString();
+
+            // Tells you how often we should do a cookie sync (in days)
+            const frequencyCap = pixelSettings.frequencyCap;
+
+            // Url for cookie sync pixel
+            const pixelUrl = replaceAmpWithAmpersand(pixelSettings.pixelUrl);
+
+            // TODO: Document requirements for redirectUrl
+            const redirectUrl = pixelSettings.redirectUrl
+                ? replaceAmpWithAmpersand(pixelSettings.redirectUrl)
+                : null;
+
+            const urlWithRedirect = this.combineUrlWithRedirect(
+                mpid,
+                pixelUrl,
+                redirectUrl
+            );
+
+            // TODO: Is there a historic reason for checking for previousMPID?
+            //       it does not appear to be passed in anywhere
+            if (previousMPID && previousMPID !== mpid) {
+                if (persistence && persistence[mpid]) {
+                    if (!persistence[mpid].csd) {
+                        persistence[mpid].csd = {};
+                    }
+                    self.performCookieSync(
+                        urlWithRedirect,
+                        moduleId,
+                        mpid,
+                        persistence[mpid].csd,
+                        filteringConsentRuleValues,
+                        mpidIsNotInCookies,
+                        requiresConsent
+                    );
+                }
+                return;
+            } else {
+                if (!persistence || !persistence[mpid]) {
+                    return;
+                }
+
+                if (!persistence[mpid].csd) {
+                    persistence[mpid].csd = {};
+                }
+
+                const lastSyncDateForModule = persistence[mpid].csd[moduleId]
+                    ? persistence[mpid].csd[moduleId]
+                    : null;
```

Comment:
not sure why this was every ternary if it's setting to `null` if `undefined`:
```suggestion
                const lastSyncDateForModule = persistence[mpid].csd[moduleId] || null;
```